cst438_register/src/main/java/com/cst438/service/GradebookServiceREST.java

```diff
@@ -8,20 +8,32 @@
 
 public class GradebookServiceREST extends GradebookService {
 
-	private RestTemplate restTemplate = new RestTemplate();
-
-	@Value("${gradebook.url}")
-	String gradebook_url;
-	
-	public GradebookServiceREST() {
-		System.out.println("REST grade book service");
-	}
-
-	@Override
-	public void enrollStudent(String student_email, String student_name, int course_id) {
-		
-		//TODO  complete this method in homework 4
-		
-	}
-
-}
+   private RestTemplate restTemplate = new RestTemplate();
+
+   @Value("${gradebook.url}")
+   String gradebook_url;
+   
+   public GradebookServiceREST() {
+      System.out.println("REST grade book service");
+   }
+
+   @Override
+   public void enrollStudent(String student_email, String student_name, int course_id) {
+      
+      //TODO  complete this method in homework 4
+      EnrollmentDTO enrollment = new EnrollmentDTO();
+      enrollment.studentEmail = student_email;
+      enrollment.studentName = student_name;
+      enrollment.course_id = course_id;
+      
+      // Debug
+      System.out.println("Post to Gradebook: " + enrollment);
+      
+      // Response from server
+      EnrollmentDTO response = restTemplate.postForObject(gradebook_url + "/enrollment", enrollment, EnrollmentDTO.class);
```

Comment:
You could add a check if response == null here and handle it accordingly. I didn't do this in mine, just sayin'.